packages/apptive_grid_heinzelmen/lib/src/invitation_dialog/invitation_dialog.dart

```diff
@@ -0,0 +1,344 @@
+import 'dart:convert';
+
+import 'package:apptive_grid_core/apptive_grid_core.dart';
+import 'package:apptive_grid_heinzelmen/src/invitation_dialog/invitation_l10n_de.dart';
+import 'package:apptive_grid_heinzelmen/src/invitation_dialog/invitation_l10n_en.dart';
+import 'package:apptive_grid_heinzelmen/src/invitation_dialog/invitation_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:http/http.dart';
+
+export 'package:apptive_grid_heinzelmen/src/invitation_dialog/invitation_localization.dart';
+
+/// Shows a Dialog to invite/add someone to a [Space]
+/// [space] is the [Space] that should be shared. [Space.links] needs to contain the Link for [type]
+/// Also [type] needs to be either [ApptiveLinkType.invite] or [ApptiveLinkType.addSpace]
+/// Note: Even when using [ApptiveLinkType.addSpace] the Ui will display Strings pointing to the user being invited to a space
+/// This can be prevented by using a custom [localization]
+///
+/// It is also possible to define which [roles] can be added. This can't be empty. Also if it only contains one [Role] the selection box is not shown
+/// This will return a Future<bool> which will return if one or more invites where send
+Future<bool> showSpaceInvitationDialog({
+  required BuildContext context,
+  required Space space,
+  List<Role> allowedRoles = Role.values,
+  ApptiveLinkType type = ApptiveLinkType.invite,
+  InvitationLocalization? localization,
+}) async {
+  assert(type == ApptiveLinkType.invite || type == ApptiveLinkType.addShare);
+  assert(space.links.containsKey(type));
+  assert(allowedRoles.isNotEmpty);
+
+  final l10n = localization ??
+      (Localizations.localeOf(context).languageCode == 'de'
+          ? const InvitationLocalizationDE()
+          : const InvitationLocalizationEn());
+
+  TextEditingController emailController = TextEditingController();
+  ValueNotifier<Role> roleNotifier = ValueNotifier(allowedRoles.first);
+  ValueNotifier<bool> loading = ValueNotifier(false);
+  ValueNotifier<dynamic> errorNotifier = ValueNotifier(null);
+  bool dialogResult = false;
+  return showDialog(
+    context: context,
+    barrierDismissible: false,
```

Comment:
I made this to prevent users from closing the dialog (on accident) to prevent the dialog from being closed while the request is running.
I guess we could allow this as long as we move confirmation snackbar display logic in the callback of the button to still get the confirmation after the dialog was closed (errors would be ignored but I guess that is ok)